lib/types/addresstrie_test.go

```diff
@@ -0,0 +1,144 @@
+package types
+
+import (
+	"encoding/json"
+	"net"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAddressTrie(t *testing.T) {
+	t.Parallel()
+
+	at := NewAddressTrie(map[string]HostAddress{
```

Comment:
Quick uninformed question, do we want to support compound wildcards `*.hello.*`. Does it even make sense? I'm not sure. If this is nonsensical or has not added value, please disregard this 👍🏻

Reply:
Ah, one use case I think a second asterisk as suffix would be convenient is for ports. You can already specify a port along with the hostname, e.g. `'test.k6.io:443': '1.2.3.4:8443'`. So maybe someone would find it useful to override only specific ports, like `'*.k6.io:24*': '1.2.3.4:8443'`. But I think this would be very niche and complicate the implementation, so let's defer to YAGNI for the second asterisk.

Reply:
IIRC in #1532, only one wildcard is chosen. that's why I implemented it like that. YAGNI is my best friend 😆

Reply:
Fair enough, thanks a lot for the pointers folks. If everyone's okay with it, let's disregard my question indeed, and reevaluate it a later point if we do end up needing it then 👍🏻 🙇🏻